packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,106 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import * as iam from '../../aws-iam';
+import * as kinesis from '../../aws-kinesis';
+import { IResource, Resource } from '../../core';
+
+/**
+ * Represents the endpoints available for targetting within a realtime log config resource
+ */
+export abstract class Endpoint {
+  public static fromKinesisStream(stream: kinesis.IStream, role: iam.IRole): Endpoint {
+    return new (class extends Endpoint {
+      public _renderEndpoint() {
+        return {
+          kinesisStreamConfig: {
+            roleArn: role.roleArn,
+            streamArn: stream.streamArn,
+          },
+          streamType: 'Kinesis',
+        };
+      }
+    });
+  }
+
+  private constructor() {}
+
+  /**
+  * @internal
+  */
+  public abstract _renderEndpoint(): any;
+}
+
+/**
+ * Represents Realtime Log Configuration
+ */
+export interface IRealtimeLogConfig extends IResource {
+  /**
+   * The name of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigName: string;
+  /**
+   * The arn of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigArn: string;
+}
+
+/**
+ * Properties for defining a RealtimeLogConfig resource.
+ */
+export interface RealtimeLogConfigProps {
+  /**
+   * The unique name of this real-time log configuration.
+   */
+  readonly name: string;
```

Comment:
Can be made optional by using `Names.uniqueId()` as default

Reply:
Changed to make `name` optional